### PR TITLE
UITabContainer and UILayout enhancements

### PR DESCRIPTION
**Pull request by JimMarlowe**
This PR has usability enhancements for UITabContainer and UILayout.

The UITabContainer has a couple of methods added to make this class usable. And all the method arguments will propagate thru the scripting languages.

```
    int GetCurrentPage(); /// returns the current page number
    bool DeletePage( int page ); /// deletes a tab + page, returns true if successful
    void AddTabPageFile ( const String &tabname, const String &layoutFile ); /// adds a tab + page from file
    void AddTabPageWidget ( const String &tabname, UIWidget *widget ); /// adds a tab + page widget(s)

```

The UI/TBLayout has a method added to program the 5 main functions in a singe call. These functions are interrelated, so if you change one, the chances are very good you'll be changing more of them. This cuts to this annoying chase.

`    void SetLayoutConfig ( const String &settings );  /// program bulk layout functionality 
`

    The string that used has at least 5 alphanumeric characters to set each of the afflicted functions.  If the input string is less than 5 characters it will not program any of the fields. Text above the 5th character ignored.  A '-' character in any field will not program that entry. Any character used which is not an allowed characters or a '-' will result in the default setting to be used.

    Character [0]  can be  'X' or 'Y'  for UI_AXIS = X(D), Y
    Character [1]  can be  'A'|'G'|'P' for UI_LAYOUT_SIZE = Available, Gravity(D), Preferred
    Character [2]  can be  'C'|'G'|'L'|'R'  for UI_LAYOUT_POSITION = Center(D), Gravity, LeftTop, RightBottom
    Character [3]  can be  'A'|'G'|'P'  for UI_LAYOUT_DISTRIBUTION = Available, Gravity, Preferred(D)
    Character [4]  can be  'C'|'L'|'R'  for UI_LAYOUT_DISTRIBUTION_POSITION, Center(D), LeftTop, RightBottom

A token was added to the layout file for the TBLayout, use  `TBLayout: config: "XGCPC" ` for programming.

### CODE REVIEW

File: Source/Atomic/UI/UILayout.h [hunk 1]
```diff
@@ -130,6 +130,17 @@ class ATOMIC_API UILayout : public UIWidget
...
+    /// character [0] = 'X' or 'Y'  for UI_AXIS = X(D), Y
+    /// character [1] = 'A'|'G'|'P' for UI_LAYOUT_SIZE = Available, Gravity(D), Perferred
+    /// character [2] = 'C'|'G'|'L'|'R'  for UI_LAYOUT_POSITION = Center(D), Gravity, LeftTop, RightBottom
+    /// character [3] = 'A'|'G'|'P'  for UI_LAYOUT_DISTRIBUTION = Available, Gravity, Perferred(D)
+    /// character [4] = 'C'|'L'|'R'  for UI_LAYOUT_DISTRIBUTION_POSITION, Center(D), LeftTop, RightBottom
+    /// A '-' character in any field will not program that entry.
+    /// Any character used that is not an allowed characters or a '-' will result in the default setting to be used.
+    /// Any text in the string above character 5 is ignored.
+    /// If the input string is less than 5 characters it will not program any of the fields.
+    void SetLayoutConfig ( const String &settings );
```
Comment (JoshEngebretson) [inline Source/Atomic/UI/UILayout.h]: This is really clever, reminds me of linux permissions, very 🐫 
aws

File: Source/Atomic/UI/UITabContainer.h [hunk 2]
```diff
@@ -44,6 +44,11 @@ class ATOMIC_API UITabContainer : public UIWidget
     UIWidget* GetCurrentPageWidget();
 
     UILayout* GetTabLayout();
+    
+    int GetCurrentPage(); /// returns the current page number
```
Comment (JoshEngebretson) [inline Source/Atomic/UI/UITabContainer.h]: Excellent API additions

File: Source/ThirdParty/TurboBadger/tb_widgets_reader.cpp [hunk 3]
```diff
@@ -263,6 +263,12 @@ void TBLayout::OnInflate(const INFLATE_INFO &info)
             ld = LAYOUT_DISTRIBUTION_POSITION_RIGHT_BOTTOM;
         SetLayoutDistributionPosition(ld);
     }
+
+// ATOMIC BEGIN
+    if (const char *layoutConfig = info.node->GetValueString("config", nullptr))
+        SetLayoutConfig(layoutConfig);
+// ATOMIC END
```
Comment (JoshEngebretson) [inline Source/ThirdParty/TurboBadger/tb_widgets_reader.cpp]: Much power, much brief code, very good.

**Comment by JimMarlowe** [comment]
The new APIs will help with the other scripting languages. I pushed a version of the periodic program up to https://github.com/JimMarlowe/AtomicExamples/tree/JM-PERIODIC/UIPeriodicTable so you can see whats-what, though its still not done.

**Pull request merged**